### PR TITLE
Manifest v3 migration

### PR DESCRIPTION
**Pull request by Mikescops**
As proposed in #22 here is a pull request to support the new manifest v3.

A few considerations:

- the update is a breaking change with v2
- the background script is now a service worker and should be placed at the root of the extension. As per [this thread](https://groups.google.com/a/chromium.org/g/chromium-extensions/c/16yaQeg07t4?pli=1) I used the `importScripts()` method to reuse most of the existing implementation.

Work left to do:

- [ ] I didn't update the tests for the snapshot, I'm not used to this system.
- [ ] Make sure the livereload is still working
- [ ] Add some documentation for people migrating to v3 with neutrino-webextension

### CODE REVIEW

**Comment by crimx** [comment]
Thanks for the input! 

> the update is a breaking change with v2

Can this be done intelligently? Like switching to different implementation base on the `manifest_version` field.

**Comment by Mikescops** [comment]
I think we can do so yes, let's try.

**Comment by Mikescops** [comment]
How do you like something like this? :arrow_up:

**Comment by Mikescops** [comment]
Considering that Firefox may not be compatible with v3 anytime soon, I recommend adding an explanation to have 2 different `.neutrinorc.js` configuration with different manifest so that it's easy to build for v2 and v3 in a same project.

File: .neutrinorc.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ module.exports = {
       popup: {
         entry: 'src/popup',
         webext: {
-          type: 'browser_action'
+          type: 'action'
```
Comment (crimx) [inline .neutrinorc.js:24]: Why this is renamed?

File: test/specs/WexExtManifestPlugin.spec.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ describe('WexExtManifestPlugin', () => {
     jest.doMock(
       path.resolve(options.manifest, 'common.manifest'),
       () => ({
-        mainfest_version: 2,
+        manifest_version: 3,
```
Comment (crimx) [inline test/specs/WexExtManifestPlugin.spec.js:61]: There should be another `.spec.js` file for v3.

**Review by crimx — CHANGES_REQUESTED** [review]
The implementation looks so cool! Can you just copy everything inside `test/` to `test/manifestV3/` and make changes there instead? I'll take care of the rest.

File: .neutrinorc.js (hunk 1 above)
Comment (Mikescops) [reply .neutrinorc.js:24]: Google merged browser_action and page_action in v3, see https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#action-api-unification

File: test/specs/WexExtManifestPlugin.spec.js (hunk 2 above)
Comment (Mikescops) [reply test/specs/WexExtManifestPlugin.spec.js:61]: yes sure, i'm going to have two different tests files

**Comment by Mikescops** [comment]
> 
> 
> The implementation looks so cool! Can you just copy everything inside `test/` to `test/manifestV3/` and make changes there instead? I'll take care of the rest.

I'd recommend not having a /test folder and using the branch system to separate the new version 👍 

let say:
- until release `master`branch is the current v2  and `beta` branch is the v3+v2
- after release `v2` branch will be the v2 and `master` branch will be v3+v2

that way we don't add noise to the repository and we can backport any potential hotfix of the v2

File: .neutrinorc.js (hunk 1 above)
Comment (crimx) [reply .neutrinorc.js:24]: I think we should keep both for v2 support.

**Comment by crimx** [comment]
I was going to do something like

```
/test/
  manifestV2/
  manifestV3/
```

so that I can test both of them. What's the downside of this approach?

**Comment by Mikescops** [comment]
Ah sorry, only for the tests, ok, I misunderstood, i thought you wanted to duplicate the whole code, in a subfolder for the v3

File: .neutrinorc.js (hunk 1 above)
Comment (Mikescops) [reply .neutrinorc.js:24]: that's the case, it supports both :)

**Comment by Mikescops** [comment]
Took me a while but I finally figured out how to build two presets at the same time.
So now:
- tests are in separate folder
- it's possible to build v2 and v3 at the same time with different configurations (for Firefox in v2 on one side and Chrome in v3 on the other for instance)

**Comment by crimx** [comment]
LGTM! Thanks for the PR!

**Pull request merged**